src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -34,16 +34,29 @@ class DelegatingLoader extends BaseDelegatingLoader
     /**
      * Constructor.
      *
+     * Ability to pass a LoggerInterface instance as second argument will be removed in 3.0.
+     *
      * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
      * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, $resolver = null, $r = null)
     {
         $this->parser = $parser;
-        $this->logger = $logger;
 
-        parent::__construct($resolver);
+        if ($resolver instanceof LoggerInterface) {
+            $this->logger = $resolver;
+
+            @trigger_error('Passing a LoggerInterface instance a second argument of the '.__METHOD__.' method is now deprecated and will not be supported anymore in 3.0.', E_USER_DEPRECATED);
```

Comment:
This is ensured by the parent constructor call which does not accept a null value. Do you think doing this check here is required too ?